lib/utils/get-role.js

```diff
@@ -0,0 +1,93 @@
+const {getProp, getPropValue} = require('jsx-ast-utils')
+const {elementRoles} = require('aria-query')
+const {getElementType} = require('./get-element-type')
+const ObjectMap = require('./object-map')
+
+// Clean-up `elementRoles` from `aria-query`
+const elementRolesMap = new ObjectMap()
+for (const [key, value] of elementRoles.entries()) {
+  // - Remove unused `constraints` key
+  delete key.constraints
```

Comment:
Is this line still necessary?

Reply:
Oooh nice catch! It looks like removing this constraints can result in incorrect mappings in some scenarios... 

For example, `elementRoles` includes:
```
[{"constraints": ["ancestor table element has table role"], "name": "td"}, ["cell"]]
[{"constraints": ["ancestor table element has grid role", "ancestor table element has treegrid role"], "name": "td"}, ["gridcell"]]
```

If we remove the constraints key, we're left with:

```
[{"name": "td"}, ["cell"]]
[{"name": "td"}, ["gridcell"]]
```

Since we cannot always accurately parse out the role of an ancestor in a linter context, I think we'd want to return undefined instead of potentially returning a false positive role.

I will follow-up!

Reply:
I deleted this line in https://github.com/github/eslint-plugin-github/pull/448/commits/8fb64196e8aa8df8819facda7284100f2577a30a!

However, there were two scenarios where it seemed appropriate to assume the role despite the constraints defined in `aria-query`. I opted to hard-code those with a comment on why! You can see the commit for more information.

I've added additional tests in https://github.com/github/eslint-plugin-github/pull/448/commits/aaca700b73ca9bc778bf254defc78eecd7e6e1e9#diff-a6b7fa0d4a2be411ccfcc6e6db5466c135100a1a045599f5c211bb91c37b839dR180-R190.